src/aggregator/client/config.go

```diff
@@ -75,70 +80,109 @@ func (c *Configuration) NewClient(
 	if err != nil {
 		return nil, err
 	}
-	return NewClient(opts), nil
+	return NewClient(opts)
```

Comment:
Think this needs to `return NewClient(opts), nil` still

Reply:
`NewClient` now returns an error so can just directly return the result:
```go
func NewClient(opts Options) (Client, error)
```